pkg/platform/kube/platform.go

```diff
@@ -801,13 +821,31 @@ func (p *Platform) CreateAPIGateway(ctx context.Context,
 
 // UpdateAPIGateway will update a previously existing api gateway
 func (p *Platform) UpdateAPIGateway(ctx context.Context, updateAPIGatewayOptions *platform.UpdateAPIGatewayOptions) error {
+	// get existing api gateway
 	apiGateway, err := p.consumer.NuclioClientSet.NuclioV1beta1().
 		NuclioAPIGateways(updateAPIGatewayOptions.APIGatewayConfig.Meta.Namespace).
 		Get(ctx, updateAPIGatewayOptions.APIGatewayConfig.Meta.Name, metav1.GetOptions{})
 	if err != nil {
 		return errors.Wrap(err, "Failed to get api gateway to update")
 	}
 
+	// restore existing config
+	var restoredAPIGatewayConfig *platform.APIGatewayConfig
+	if restoredAPIGatewayConfig, err = p.apiGatewayScrubber.RestoreAPIGatewayConfig(ctx, &platform.APIGatewayConfig{
```

Comment:
This step can be skipped if masking is disabled or there's nothing to restore...

Reply:
@TomerShor not really, because we can't be sure that platform configuration wasn't changed since api gateway was created, so I'd try to restore it anyways.

Reply:
That's why we have `HasScrubbedConfig` in the Abstract scrubber, that can be relevant to AGW as well

Reply:
Make sense, will add!